bind: fix link-local addresses scoped with interface names

Bill-Sommerfeld opened the pull request:
If bind's configure can't find if_nametoindex(), it only lets you use interface ids (fe80::1%5) instead of interface names (like fe80::1%wan0); pass LIBS=-lsocket to configure and it will find the function in -lsocket.

AndWac commented on components/network/bind/Makefile:
```diff
@@ -83,6 +83,7 @@ CONFIGURE_OPTIONS += --with-tuning=default
+CONFIGURE_OPTIONS += LIBS="-lsocket"
 
 # Install.
 #
```
When you change a package you either need to change COMPONENT_VERSION or increment/set COMPONENT_REVISION. So please add
COMPONENT_REVISION= 1

Bill-Sommerfeld replied:
Done.

AndWac replied:
Where? I cannot see it in the Makefile...

mtelka replied:
Sorry, not done.

Please also make sure you ran following make targets at least once, all of them passed and there is nothing changed in the bind component directory: `sample-manifest`, `publish`, `test`, `REQUIRED_PACKAGES`.  Thank you.

Bill-Sommerfeld replied:
Arrgh, miss a `git add` and I look like an idiot.

I'm chasing down a test issue (tests work fine when I cd into the build tree, but when run from the component directory, they all fail with errors resembling:
```
./aes_test: line 43: libtool_execute_magic: unbound variable
FAIL aes_test (exit status: 1)
```
 it looks like it's the result of SHELLOPTS being exported rather than my change; I see many component makefiles have a `unexport SHELLOPTS` and will be adding one to bind's.

Bill-Sommerfeld committed:
bind: fix link-local addresses scoped with interface names

If bind's configure can't find if_nametoindex(), it only lets you use
numeric interface ids (fe80::1%5) instead of interface names
(fe80::1%wan0) to scope link-local addresses.  At the suggestion of
Alan Coopersmith, update configure.ac to search for if_nametoindex in
libsocket.

Also along for the ride, to fix tests:

1) unexport SHELLOPTS to avoid libtool wrapper failures

2) Clear LD_B_DIRECT; direct binding of bind's shared libraries breaks several
of the self tests.

3) Fix how we set the timezone to GMT in one of the tests.

4) When adding test ipv4 addresses to lo0, use a /32 prefix len so that
it doesn't default to a /8.
